exonum/src/runtime/dispatcher.rs

```diff
@@ -142,6 +143,16 @@ impl RuntimeEnvironment for Dispatcher {
         }
     }
 
+    fn state_hashes(&self, snapshot: &dyn Snapshot) -> Vec<(ServiceInstanceId, Vec<Hash>)> {
+        let mut hashes = Vec::new();
+
+        for (_, runtime) in &self.runtimes {
+            hashes.append(&mut runtime.state_hashes(snapshot));
+        }
+
+        hashes
+    }
```

Comment:
Or you can re-write it for something like this:
```rust
fn state_hashes(&self, snapshot: &dyn Snapshot) -> Vec<(ServiceInstanceId, Vec<Hash>)> {
        self.runtimes
            .iter()
            .map(|(_, runtime)| runtime.state_hashes(snapshot))
            .flatten()
            .collect::<Vec<_>>()
}
```

Reply:
Hmm, dunno. I like plain&dumb style here more. However, if you insist...

Reply:
Not really but IMO looks better.